src/components/Form/Radiobutton.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { Component, ReactNode } from 'react';
+import classnames from 'classnames';
+import { HtmlInput, HtmlLabel, HtmlSpan, HtmlDiv } from '../../reset';
+import { idGenerator } from '../../utils/uuid';
+import { logger } from '../../utils/logger';
+import {
+  RadiobuttonGroupConsumer,
+  RadiobuttonGroupProviderState,
+} from './RadiobuttonGroup';
+
+const baseClassName = 'fi-radiobutton';
+
+const radiobuttonClassNames = {
+  container: `${baseClassName}_container`,
+  input: `${baseClassName}_input`,
+  label: `${baseClassName}_label`,
+  hintText: `${baseClassName}_hintText`,
+};
+
+type RadiobuttonVariant = 'small' | 'large';
+
+export interface RadiobuttonProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /**
+   * Label element content
+   */
+  children?: ReactNode;
+  /** Group multiple Radiobuttons to be inside same selection group */
+  name?: string;
+  /** Unique value to be used inside Radiobutton group for this item */
+  value?: string;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated using uuid
+   * @default uuidV4
+   */
+  id?: string;
+  /**
+   * Hint text to be displayed under the label.
+   */
+  hintText?: string;
+  /** Disable Radiobutton. Value won't be included when submitting */
+  disabled?: boolean;
+  /**
+   * 'small' | 'large'
+   * @default small
+   */
+  variant?: RadiobuttonVariant;
+  radiobuttonGroup?: boolean;
+  consumer?: RadiobuttonGroupProviderState;
+  checked?: boolean;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
These props are for using the Radiobutton in group and therefore do not have public JSDoc. We should "enforce" of using RadiobuttonGroup instead of manually doing the same.

Reply:
onChange is not documented, but yet it is used in the examples. I'd either document the prop or remove it from the example.

Reply:
Btw. Should there be an onChange event for the group? And about the type, is the whole event necessary here, or could we just pass the new value if it really changes?

Reply:
Currently there is no `onChange` prop for group. But it is good idea to be added.

One could possibly use the existing https://github.com/vrk-kpa/suomifi-ui-components/blob/5a9f7a28d5748b56a3f92007fdb12fceb3ca7949/src/components/Form/RadiobuttonGroup.tsx#L47-L49

with the implementation. It is used to handle the case when one of the radios are selected in the
https://github.com/vrk-kpa/suomifi-ui-components/blob/5a9f7a28d5748b56a3f92007fdb12fceb3ca7949/src/components/Form/RadiobuttonGroup.tsx#L98-L100

Reply:
I think it would make using these easier since only one listener would be sufficient.

Reply:
Unless it messes the controlled state somehow when used mixed with the ones for radiobutton 😓

Reply:
Documentation updated.

Reply:
And onChange prop for the group added along with existence check for value in group props to prevent changing state accidentally in controlled state.